### PR TITLE
Migrate vrrp_sync_group to custom resource

### PR DESCRIPTION
**Pull request by Xorima**
### Description

Migrate this hwrp over to custom resources

Signed-off-by: Jason Field <jason@avon-lea.co.uk>

### Issues Resolved

[List any existing issues this PR resolves]

### Check List
- [ ] All tests pass. See https://github.com/sous-chefs/grafana/blob/master/TESTING.md
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: documentation/keepalived_vrrp_sync_group.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+[back to resource list](https://github.com/sous-chefs/keepalived#resources)
```
Comment (damacus) [inline documentation/keepalived_vrrp_sync_group.md]: We probably want a list of resources under resources here

File: documentation/keepalived_vrrp_sync_group.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# keepalived_vrrp_sync_group
+
+The `keepalived_vrrp_sync_group` resource can be used to configure VRRP Sync Groups (groups of resources that fail over together).
+
+More information available at <https://www.keepalived.org/manpage.html>
+
+## Actions
+
+`:create`
+`:delete`
```
Comment (damacus) [inline documentation/keepalived_vrrp_sync_group.md]: List?

File: libraries/keepalived.rb [hunk 3]
```diff
@@ -46,11 +47,11 @@ module Notify
...
-  module VrrpSyncGroup
-    OPTIONS ||= Notify::OPTIONS.merge(
-      group: { kind_of: Array, required: true }
-    ).freeze
-  end
+  # module VrrpSyncGroup
+  #   OPTIONS ||= Notify::OPTIONS.merge(
+  #     group: { kind_of: Array, required: true }
+  #   ).freeze
+  # end
```
Comment (damacus) [inline libraries/keepalived.rb]:
```suggestion
```

File: libraries/keepalived.rb [hunk 4]
```diff
@@ -35,6 +35,7 @@ module Keepalived
     smtp_check
     misc_check
   ).map { |r| "keepalived_#{r}".to_sym }.freeze
+  # vrrp_sync_group
```
Comment (damacus) [inline libraries/keepalived.rb]:
```suggestion
```

File: libraries/keepalived.rb [hunk 5]
```diff
@@ -23,7 +23,7 @@ module Keepalived
   HEALTH_PATH ||= "#{ROOT_PATH}/checks.d".freeze
   RESOURCES ||= %w(
     config
-    vrrp_sync_group
+
```
Comment (damacus) [inline libraries/keepalived.rb]: Probably want to remove this random line

File: libraries/resources.rb [hunk 6]
```diff
@@ -47,22 +47,22 @@ def to_conf
...
+
+    #   def to_conf
+    #     cfg = ["vrrp_sync_group #{name} {"]
+    #     cfg << Keepalived::Helpers.conf_string(
+    #       self, Keepalived::VrrpSyncGroup::OPTIONS
+    #     )
+    #     cfg << '}'
+    #     cfg.join("\n\t")
+    #   end
+    # end
```
Comment (damacus) [inline libraries/resources.rb]: Still want this at all?

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -42,8 +42,11 @@ jobs:
         os:
           - 'amazonlinux'
           - 'debian-9'
+          - 'debian-10'
           - 'ubuntu-1604'
+          - 'ubuntu-1804'
```
Comment (tas50) [inline .github/workflows/ci.yml]: Prob not for this one, but you might want to get 20.04 in here

File: documentation/keepalived_vrrp_sync_group.md (hunk 1 above)
Comment (Xorima) [reply documentation/keepalived_vrrp_sync_group.md]: That will get done as part of the final merge to master (This is going to a staging branch for now)

File: .github/workflows/ci.yml (hunk 7 above)
Comment (Xorima) [reply .github/workflows/ci.yml]: Done as well
Moved amazonlinux to amazonlinux-2

**Pull request merged**